iceoryx_posh/include/iceoryx_posh/popo/listener.hpp

```diff
@@ -197,21 +203,31 @@ class Listener
         uint64_t indicesInUse() const noexcept;
 
         using LoFFLi = concurrent::LoFFLi;
-        LoFFLi::Index_t
-            m_loffliStorage[LoFFLi::requiredIndexMemorySize(MAX_NUMBER_OF_EVENTS_PER_LISTENER) / sizeof(uint32_t)];
+        LoFFLi::Index_t m_loffliStorage[LoFFLi::requiredIndexMemorySize(Capacity) / sizeof(uint32_t)];
         LoFFLi m_loffli;
         std::atomic<uint64_t> m_indicesInUse{0U};
     } m_indexManager;
 
 
     std::thread m_thread;
-    concurrent::smart_lock<Event_t, std::recursive_mutex> m_events[MAX_NUMBER_OF_EVENTS_PER_LISTENER];
+    concurrent::smart_lock<internal::Event_t, std::recursive_mutex> m_events[Capacity];
```

Comment:
Regarding my comment with the detail namespace: it appears we already have something like a detail namespace here (internal) and also some kind of implementation class `ListenerImpl`.